### PR TITLE
Keep active tab state in one place (manager)

### PR DESCRIPTION
**Pull request by notlaforge**
With this change TabsBase objects will retain a reference to the TabsManager they belong to, so they can ask it if they're the
active tab or not.

This removes the possibility for bugs like activating an unmanaged tab, or having more than one active tab, etc.

[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/683)

### CODE REVIEW

File: src/Explorer/Tabs/TabsManager.ts [hunk 1]
```diff
@@ -4,23 +4,25 @@ import Explorer from "../Explorer";
...
-  public activeTab: ko.Observable<TabsBase>;
+  private idx = ko.observable(0);
+  public openedTabs = ko.observableArray<TabsBase>([]);
+  public activeTab = ko.pureComputed(() => this.openedTabs()[this.idx()]);
 
   constructor() {
-    this.openedTabs = ko.observableArray<TabsBase>([]);
-    this.activeTab = ko.observable<TabsBase>();
+    this.openedTabs.subscribe(
+      (log) => log.forEach(({ value: tab, status }) => tab.manager(status === "deleted" ? undefined : this)),
```
Comment (southpolesteve) [inline src/Explorer/Tabs/TabsManager.ts:13]: Extra log left behind?

File: src/Explorer/Tabs/TabsBase.ts [hunk 2]
```diff
@@ -32,6 +33,7 @@ export default class TabsBase extends WaitsForTemplateViewModel {
   public isExecutionError: ko.Observable<boolean>;
   public isExecuting: ko.Observable<boolean>;
   public pendingNotification?: ko.Observable<DataModels.Notification>;
+  public manager = ko.observable<TabsManager>(undefined);
```
Comment (southpolesteve) [inline src/Explorer/Tabs/TabsBase.ts:36]: Does this need to be wrapped in observable? No problem if yes just would be nice to skip if not needed

**Review by southpolesteve — APPROVED** [review]

File: src/Explorer/Tabs/TabsManager.ts (hunk 1 above)
Comment (notlaforge) [reply src/Explorer/Tabs/TabsManager.ts:13]: No, it's the array change*log*. I'll give it a better name.

File: src/Explorer/Tabs/TabsBase.ts (hunk 2 above)
Comment (notlaforge) [reply src/Explorer/Tabs/TabsBase.ts]: Thanks, I made some changes and it is no longer an Observable, nice catch.

**Pull request merged**